relentless/optimize/objective.py

```diff
@@ -165,3 +181,150 @@ def design_variables(self):
         :class:`ObjectiveFunction` for which the result was constructed, mapped
         to the value of the variables at the time the result was constructed."""
         return self._design_variables
+
+class RelativeEntropy(ObjectiveFunction):
+    r"""Relative entropy.
+
+    The relative entropy quantifies how different two probability distributions are.
+    This can be applied to the probability distributions of a target ensemble
+    and the ensemble of a simulated molecular system, which is parametrized
+    by some pair potential functions.
+
+    The value of the relative entropy is not determined. The gradient of the relative
+    entropy is equivalent to the gradient of the likelihood of the design variables
+    :math:`\mathbf{x}` given the target ensemble :math:`\mathbf{T}`. Given
+    the rdfs of the simulation and target ensembles, :math:`g(r|\mathbf{x})` and :math:`g_{tgt}(r)`,
+    respectively, the total pair potential energy, the gradient is:
+
+    .. math::
+
+        \nabla_\mathbf{x} L\left(\mathbf{x}|\mathbf{T}\right) = 2\pi N\rho\int{dr r^2 \left(g(r|\mathbf{x})-g_{tgt}(r)\right)\beta u(r|\mathbf{x})}
+
+    where :math:`\beta` is the thermodynamic beta, :math:`N` is the particular number,
+    and :math:`\rho=N/V` is the number density, all for the target ensemble.
```

Comment:
Here is a rewrite of these docs. I also propose that we make the normalization of S an option of the class so that you can still get the extensive version if you want. I propose adding an option `intensive` that defaults to True, but it could also be called `extensive`, and I don't have a strong preference what the default value is.

Also, somebody should check my math on the equations :)

```suggestion
    The value of the relative entropy is not readily determined in molecular simulations,
    so this :class:`ObjectiveFunction` does not return a value.

    However, its gradient with respect to design variables **x** is much easier to compute.
    Currently, the :class:`RelativeEntropy` objective function supports
    :class:`~relentless.potential.pair.PairPotential` interactions. These interactions
    are characterized by an :class:`~relentless.ensemble.RDF` for each pair
    of interacting types in each :class:`~relentless.ensemble.Ensemble`,
    :math:`g_{ij}(r)`. The gradient of :class:`S_{\rm rel}` is then:

    .. math::

        \nabla_\mathbf{x} S_{\rm rel} = \frac{1}{2} \sum_{i,j} \int{dr 4\pi r^2 \left[\frac{\beta N_i N_j}{V} g_{ij}(r)- \frac{\beta_0 N_{i,0} N_{j,0}}{V_0} g_{ij,0}(r)\right] \nabla_\mathbf{x} u_{ij}(r)}

    where :math:`\beta = 1/(k_{\rm B}T)`, :math:`N_i` is the number of particles of
    type *i*, *V* is the volume, and :math:`u_{ij}` is the pair potential in the *model*
    ensemble.
    
    The relative entropy :math:`S_{\rm rel}` is extensive as written, meaning that it
    depends on the size of the system. This can be undesirable for optimization because
    it means certain hyperparameters are system-size dependent. To circumvent this,
    use ``intensive=True`` (default) to normalize :math:`s_{\rm rel} = S_{\rm rel}/V_0`.
    The extensive relative entropy can still be computed by setting ``intensive=False``.
```